core/request.go

```diff
@@ -105,22 +105,32 @@ func (r *RequestAccessor) StripBasePath(basePath string) string {
 }
 
 // ProxyEventToHTTPRequest converts an API Gateway proxy event into a http.Request object.
-// Returns the populated request with:
-// * an additional two custom headers for the stage variables and API Gateway context.
-//   To access these properties use the GetAPIGatewayStageVars and GetAPIGatewayContext method of the RequestAccessor object.
-// * lambda context passed as well as APIGatewayProxyRequestContext as part of its context under different keys.
-//   Access those using GetAPIGatewayContextFromContext and GetRuntimeContextFromContext functions in this package.
+// Returns the populated http request with additional two custom headers for the stage variables and API Gateway context.
+// To access these properties use the GetAPIGatewayStageVars and GetAPIGatewayContext method of the RequestAccessor object.
 func (r *RequestAccessor) ProxyEventToHTTPRequest(req events.APIGatewayProxyRequest) (*http.Request, error) {
-	return r.RequestFromEvent(context.Background(), req)
+	httpRequest, err := r.RequestFromEvent(req)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	return AddToHeader(httpRequest, req)
+}
+
+// EventToRequestWithContext converts an API Gateway proxy event and context into an http.Request object.
+// Returns the populated http request with lambda context, stage variables and APIGatewayProxyRequestContext as part of its context.
+// Access those using GetAPIGatewayContextFromContext, GetStageVarsFromContext and GetRuntimeContextFromContext functions in this package.
+func (r *RequestAccessor) EventToRequestWithContext(ctx context.Context, req events.APIGatewayProxyRequest) (*http.Request, error) {
+	httpRequest, err := r.RequestFromEvent(req)
```

Comment:
This juxtaposed naming should be unified (EventToRequest vs RequestFromEvent).

Reply:
Agreed. Fixed.